Assets/LoggingManager/LogStore.cs

```diff
@@ -0,0 +1,276 @@
+﻿using System;
+using System.CodeDom;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using Boo.Lang.Runtime;
+using UnityEngine;
+
+
+public enum LogType
+{
+    Normal,
+    Meta
+}
+
+public class LogStore : MonoBehaviour
+{
+
+    private SortedDictionary<string, List<string>> logs;
+    private StringBuilder logString;
+    public string Label { get; set; }
+
+    private LogType logType;
+    private bool logCommonColumns;
+
+    private int nbLines;
+    private bool createStringOverTime;
+    private StringBuilder currentLineLogged;
+    private SortedDictionary<string, string> currentLogs;
+    private const string fieldSeparator = ";";
+    private const string lineSeperator = "\n";
+    private string email;
+    public string SessionId { get; set; }
+
+    public LogStore(string label, string email, bool createStringOverTime,
+        LogType logType = LogType.Normal, bool logCommonColumns = true)
+    {
+        Init(label, email, Guid.NewGuid().ToString(), createStringOverTime, logType, logCommonColumns);
+    }
+
+    public LogStore(string label, string email, string sessionID, bool createStringOverTime,
+        LogType logType = LogType.Normal, bool logCommonColumns = true)
+    {
+        Init(label, email, sessionID, createStringOverTime, logType, logCommonColumns);
+    }
+
+    private void Init(string label, string email, string sessionID, bool createStringOverTime,
+        LogType logType, bool logCommonColumns)
+    {
+        this.Label = label;
+        logs = new SortedDictionary<string, List<string>>();
+        logString = new StringBuilder();
+        currentLineLogged = new StringBuilder();
+        currentLogs = new SortedDictionary<string, string>();
+        this.createStringOverTime = createStringOverTime;
+        this.email = email;
+        SessionId = sessionID;
+        this.logType = logType;
+        this.logCommonColumns = logCommonColumns;
+        if (logType == LogType.Normal && logCommonColumns)
+        {
+            logs.Add("Timestamp", new List<string>());
+            logs.Add("Framecount", new List<string>());
+            logs.Add("SessionID", new List<string>());
+            logs.Add("Email", new List<string>());
+        }
+        else if (logType == LogType.Meta)
+        {
+            logs.Add("SessionID", new List<string>());
+            logs.Add("Email", new List<string>());
+        }
+    }
+
+    //Adds a column to the current row of the Logs
+    public void Add(string column, object data)
+    {
+        if (nbLines > 0 && !logs.Keys.Contains(column))
```

Comment:
Why do we need to check whether `nbLines` is above 0?

Reply:
If the right part of the condition is true, it means that a new header has been added, so we don't check this when logging the first line (when there is 0 lines) because this is the row that defines the headers.

Reply:
OK, could you add a comment about this above the if statement? Just so it is clear for the future.